doc/specific_iocs/plcs/OPCUA.md

```diff
@@ -0,0 +1,110 @@
+* Support for both client-server and publish-subscribe communication patterns
+* Communication protocol independent. Mappings to several communication protocols like TCP/IP, UDP/ 
+IP, WebSockets, AMQP and MQTT are specified
+* Initially successful in standardized data exchange with industrial equipment (discrete manufacturing, process manufacturing, energy) and systems for data collection and control, but now also leveraged in building automation, weighing and kitchen equipment and cloud applications
+* [Open](https://en.wikipedia.org/wiki/Open_standard) – open-source reference implementations freely available to OPC Foundation members, non members under GPL 2.0 license[2]
+* [Cross-platform](https://en.wikipedia.org/wiki/Cross-platform) – not tied to one operating system or programming language
+* [Service-oriented architecture](https://en.wikipedia.org/wiki/Service-oriented_architecture) (SOA)
+* The specification is freely available on the OPC Foundation website and is split into several parts 
+to ease implementation, but only OPC UA stack vendors need to read them, end users simply leverage 
+existing commercial and/or open-source stacks available in all popular programming languages
+
+More info can be found [here on Wikipedia for a general overview](https://en.wikipedia.org/wiki/OPC_Unified_Architecture), or for a more detailed description from [OPC Foundation](https://opcfoundation.org/about/what-is-opc/).
+
+## How is authentication handled?
+Currently, the IOC does not seem to be able to support encrypted message security policy. It does, however, support “None” security mode, and connecting with a username and password, which also appears to require sending the password encrypted with Basic256 (username and password connection works with None security mode, but does not work if there is no certificate and private key provided via the `opcuaCLientCertificate` option in the `st-common.cmd` or `st.cmd` file, which loads the IOC with some other options.  
+
+Currently, authentication configurations in Windows only seem to work with username and password. This is likely due to functionality missing from the `open62541` library, the open source library that we use in conjunction with `opcua` EPICS module. A username and password is set on the PLC itself, and those values can be read at IOC startup to authenticate, and sent via Basic256 encryption to the PLC to sign in. When implementing/installing onto a new instrument, the `client_private_key.pem`, `cert.txt` (which will need to be edited to reflect current username and password for the target PLC/server), and `OPCUA_01.cmd` should be moved from the Experiment Controls non-public share `OPCUA` folder, to the instrument's configurations area, in a new folder that should be named `opcua`. If done properly, the `opcua` EPICS module should be able to pick up the user name and password, log in to the OPC server properly, and begin a connection. 
```

Comment:
> `client_private_key.pem`, `cert.txt`

How does one acquire or generate these files? Are they something that was exported from the PLC, or something you generated?